aten/src/ATen/native/nested/NestedTensorMath.cpp

```diff
@@ -945,6 +947,26 @@ Tensor _NestedTensor_GeneralizedBMM(const Tensor& self, const Tensor& mat2) {
   return output;
 }
 
+Tensor& matmul_out_nested(const Tensor& tensor1, const Tensor& tensor2, Tensor& result) {
+  // TODO: this is a very quick and dirty implementation
+  //       should improve it to avoid the intermediate memory usage
```

Comment:
I'd expect this to be the other way around - matmul calls matmul_out for NT. Let's cover this in a follow-up PR.

When that high perf matmul kernel is upstreamed, that'll be nice.